docs/src/writing-shader-crates.md

```diff
@@ -31,40 +31,46 @@ There are two main ways to setup your shader project locally.
    flags in your cargo configuration to enable you to run `cargo build` in your
    shader crate.
 
-
 ### Using `spirv-builder`
 If you're writing a bigger application and you want to integrate SPIR-V shader
 crates to display, it's recommended to use `spirv-builder` in a build script.
 
 1. Copy the [`rust-toolchain`] file to your project. (You must use the same
-   version of Rust as `rust-gpu`.)
+   version of Rust as `rust-gpu`. Utimately, the build will fail with a nice
+   error message when you don't use the exact same version)
 2. Reference `spirv-builder` in your Cargo.toml:
     ```toml
     [build-dependencies]
-    spirv-builder = { git = "https://github.com/EmbarkStudios/rust-gpu" }
+    spirv-builder = "0.4"
     ```
-    (we currently do not publish spirv-builder on crates.io)
+    All dependent crates are published on [crates.io](https://crates.io).
 3. Create a `build.rs` in your project root.
 
 #### `build.rs`
-Paste the following into the `main` for your build script.
+Paste the following into `build.rs`
 
 ```rust,no_run
-SpirvBuilder::new(path_to_shader, target)
+use spirv_builder::{MetadataPrintout, SpirvBuilder};
+
+fn main() {
+    SpirvBuilder::new(shader_crate, target)
         .print_metadata(MetadataPrintout::Full)
-        .build()?;
+        .build()
+        .unwrap(); 
```

Comment:
Sorry for doing this again, but now that indentation is fixed, I can see a problem, instead of `.unwrap()` it should be e.g.:
```rust
fn main() -> Result<(), Box<dyn std::error::Error>> {
    SpirvBuilder::new(shader_crate, target)
        .print_metadata(MetadataPrintout::Full)
        .build()?;
    Ok(())
}
```
(either this, or using `SpirvBuilderError` directly, should be done instead of `.unwrap()`, to get a more readable message when it fails)